miio/g1vacuum.py

```diff
@@ -0,0 +1,312 @@
+    Paused = 3
+    Error = 4
+    Charging = 5
+    Go_Charging = 6
+
+class VacuumMode(Enum):
+    """Vacuum Mode"""
+    Global_clean = 1
+    Spot_clean = 2
+    Wiping = 3
+
+class WaterLevel(Enum):
+    """Water Flow Level"""
+    Level1 = 1
+    Level2 = 2
+    Level3 = 3
+
+class FanSpeed(Enum):
+    """Fan speeds, same as for ViomiVacuum."""
+    Mute = 0
+    Standard = 1
+    Medium = 2
+    High = 3
+
+class Languages(Enum):
+    Chinese = 0
+    English = 1
+
+class MopState(Enum):
+    Off = 0
+    On = 1
+
+class MovementDirection(Enum):
+    Left = 0
+    Right = 1
+    Forward = 2
+    Backward = 3
+    Stop = 4
+
+class G1Status:
+    """Container for status reports from the Mijia Vacuum G1."""
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self.data = data
+
+    @property
+    def battery(self) -> int:
+        """Battery Level."""
+        return self.data["battery"]
+
+    @property
+    def charge_state(self) -> ChargeState:
+        """Charging State."""
+        return ChargeState[ChargeState(self.data["charge_state"]).name]
+
+    @property
+    def error(self) -> Error:
+        """Error Message."""
+        return Error[Error(self.data["error"]).name]
+
+    @property
+    def state(self) -> State:
+        """Vacuum Status."""
+        return State[State(self.data["state"]).name]
+
+    @property
+    def fan_speed(self) -> FanSpeed:
+        """Fan Speed."""
+        return FanSpeed[FanSpeed(self.data["fan_speed"]).name]
+
+    @property
+    def operating_mode(self) -> VacuumMode:
+        """Operating Mode."""
+        return VacuumMode[VacuumMode(self.data["operating_mode"]).name]
+
+    @property
+    def mop_state(self) -> MopState:
+        """Mop State."""
+        return MopState[MopState(self.data["mop_state"]).name]
+
+    @property
+    def water_level(self) -> WaterLevel:
+        """Mop State."""
+        return WaterLevel[WaterLevel(self.data["water_level"]).name]
+
+    @property
+    def brush_life_level(self) -> int:
+        """Brush Life Level."""
+        return self.data["brush_life_level"]
+
+    @property
+    def brush_life_level2(self) -> int:
+        """Side Brush Life Level."""
+        return self.data["brush_life_level2"]
+
+    @property
+    def filter_life_level(self) -> int:
+        """Filter Life Level."""
+        return self.data["filter_life_level"]
+
+    @property
+    def clean_area(self) -> int:
+        """Clean Area."""
+        return self.data["clean_area"]
+
+    @property
+    def clean_time(self) -> int:
+        """Clean Time."""
+        return self.data["clean_time"]
+
+    @property
+    def total_clean_count(self) -> int:
+        """Total Clean Count."""
+        return self.data["total_clean_count"]
+
+    # @property
+    # def total_clean_area(self) -> int:
+        # """Total Clean Area."""
+        # return self.data["total_clean_area"]
+
+    def __repr__(self) -> str:
+        ret = (
+            "<VacuumStatus battery=%s, "
```

Comment:
This is not necessary, as inheriting from `DeviceStatus` handles it automatically.